### PR TITLE
Updates version to 2.0.1 and adds release notes

### PR DESCRIPTION
**Pull request by sshivanii**
Signed-off-by: Shivani Shukla <sshkamz@amazon.com>

### Description
* Updates version to 2.0.1
* Adds release notes for 2.0.1

### Issues Resolved
Contributes as pre-release step of 2.0.1 [#3139](https://github.com/opensearch-project/opensearch-build/issues/3139)

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has documentation added
- [ ] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by oeyh — APPROVED** [review]

File: release-notes/logstash-output-opensearch-release-notes-2.0.1.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## Version 2.0.1 Release Notes
+Compatible with OpenSearch 1.3.4, 2.1.0
```
Comment (dlvenable) [inline release-notes/logstash-output-opensearch-release-notes-2.0.1.md:2]: Should we say `OpenSearch 1.3+, 2.0+` or something to that effect?

Comment (sshivanii) [reply release-notes/logstash-output-opensearch-release-notes-2.0.1.md:2]: `1.3+` makes sense but we're not sure if releases `2.1+` would all compatible?

File: release-notes/logstash-output-opensearch-release-notes-2.0.1.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## Version 2.0.1 Release Notes
+Compatible with OpenSearch 1.3+, 2.1.0
```
Comment (asifsmohammed) [inline release-notes/logstash-output-opensearch-release-notes-2.0.1.md:2]: Should this be `2.0` or `2.1.0`?

**Review by asifsmohammed — APPROVED** [review]

File: release-notes/logstash-output-opensearch-release-notes-2.0.1.md (hunk 2 above)
Comment (sshivanii) [reply release-notes/logstash-output-opensearch-release-notes-2.0.1.md:2]: It's also compatible with `2.1.0` so kept it at that.

Comment (dlvenable) [reply release-notes/logstash-output-opensearch-release-notes-2.0.1.md:2]: Maybe `1.3+, 2.0+` so that users know they can use it with a 2.0.0 cluster.

**Review by dlvenable — APPROVED** [review]

**Review by asifsmohammed — APPROVED** [review]

**Pull request merged**